spot-contracts/contracts/vaults/RolloverVault.sol

```diff
@@ -217,30 +217,26 @@ contract RolloverVault is
         uint256 amount;
     }
 
-    /// @notice Burns notes and sends a proportional share vault's assets back to {msg.sender}.
+    /// @notice Burns notes and sends a proportional share of vault's assets back to {msg.sender}.
     /// @param notes The amount of notes to be burnt.
     /// @return The list of asset tokens and amounts redeemed.
     function redeem(uint256 notes) external nonReentrant whenNotPaused returns (TokenAmount[] memory) {
         uint256 totalNotes = totalSupply();
-        uint256 deployedCount_ = deployedCount();
-        uint256 earnedCount_ = earnedCount();
-        uint256 assetCount = 1 + deployedCount_ + earnedCount_;
-
-        TokenAmount[] memory redemptions = new TokenAmount[](assetCount);
+        uint256 deployedCount_ = _deployed.length();
+        uint256 assetCount = 2 + deployedCount_;
 
         // burn notes
         _burn(_msgSender(), notes);
 
         // aggregating vault assets to be redeemed
+        TokenAmount[] memory redemptions = new TokenAmount[](assetCount);
         redemptions[0].token = underlying;
         for (uint256 i = 0; i < deployedCount_; i++) {
-            redemptions[i + 1].token = deployedAt(i);
-        }
-        for (uint256 i = 0; i < earnedCount_; i++) {
-            redemptions[i + 1 + deployedCount_].token = earnedAt(i);
+            redemptions[i + 1].token = IERC20Upgradeable(_deployed.at(i));
         }
+        redemptions[deployedCount_ + 1].token = IERC20Upgradeable(perp);
```

Comment:
Oh, ok--we're assuming that the earned token is perp. Ties into the comment above, because presumably perp and underlying can't be the same.

Think it's cleaner to take the earned token as an initialization parameter, then not worry about it?

Reply:
The question if we keep track of "earned" and infer "perp", or keep track of perp and infer earned. I went back and forth and chose the latter ..

Reply:
ah got it. But at the storage level we just keep track of `perp` and the internal contract functions point to `perp` when they want to access `earned`. (which is how it currently works)

Reply:
Ah right, nvm